src/form_response_collection.py

```diff
@@ -13,47 +23,53 @@ def __init__(self):
         self.client = MongoClient(access_url)
         self.db = self.client.cidev_db
         self.form_response_collection = self.db.form_response_collection
+        self.form_response_collection.create_index('form', name='form-index')
 
-    def get_form_by_id(self, id):
-        form = self.form_response_collection.find_one({'id': id})
-        if not form:
-            raise ValueError
+    @catch_invalid_id
+    def get_response_by_id(self, rid):
+        response = self.form_response_collection.find_one(ObjectId(rid))
+        if not response:
+            raise ValueError(f'Form response with id {rid} does not exist.')
 
-        form['_id'] = str(form['_id'])
-        return form
+        response['_id'] = str(response['_id'])
+        return response
 
-    def get_all_forms(self):
-        forms = self.form_response_collection.find({})
-        print(forms)
+    def get_all_responses(self):
+        responses = self.form_response_collection.find({})
         result = []
-        for form in forms:
-            form['_id'] = str(form['_id'])
-            result.append(form)
+        for response in responses:
+            response['_id'] = str(response['_id'])
+            result.append(response)
 
         return result
 
-    def add_form(self, form):
-
-        self.form_response_collection.insert_one(form)
-        return True
-
-    def update_one_form(self, id, updates):
-        if not self.form_response_collection.find_one(id):
-            raise ValueError
-        self.form_response_collection.update_one(id, {'$set': updates})
-        return True
+    @catch_invalid_id
+    def get_responses_to_form(self, form_id):
+        responses = self.form_response_collection.find({'form': ObjectId(form_id)})
+        result = []
+        for response in responses:
+            response['_id'] = str(response['_id'])
+            result.append(response)
 
-    def delete_all_forms(self):
-        self.form_response_collection.delete_many({})
-        return True
+        return result
 
-    def delete_form_by_id(self, id):
-        form = self.form_response_collection.find_one({'id': id})
-        if not form:
-            raise ValueError
-        self.form_response_collection.delete_one({'id': id})
-        return True
+    @catch_invalid_id
+    def add_response(self, response, form_id):
```

Comment:
Decorator prints args[1], so in this case the entire response. The id to check needs to be the first parameter for the error message to be correct.